zzk/service/hostregistry.go

```diff
@@ -77,11 +78,22 @@ type HostRegistryListener struct {
 
 // InitHostRegistry initializes the host registry
 func InitHostRegistry(conn client.Connection) error {
-	err := conn.CreateDir(hostregpath())
-	if err == client.ErrNodeExists {
-		return nil
+	for {
+		done := make(chan error, 1)
+		go func() {
+			done <- conn.CreateDir(hostregpath())
+		}()
+		select {
+		case err := <-done:
+			if err == client.ErrNodeExists {
+				return nil
+			}
+			return err
+		case <-time.After(5 * time.Second):
+			glog.Errorf("Could not create host registry directory within 5 seconds. Retrying.")
```

Comment:
Yes. A request is somehow lost, and we're blocked waiting on a response that will never come. Doing a serviced reload causes the request to go out again, and things proceed.

Calling CreateDir on a node that exists is a noop. In this case, the node exists (we see it in exhibitor), but CreateDir is definitely blocked (we see it in the goroutine dump).